packages/helper-image/test/unit/spec/index.js

```diff
@@ -60,76 +94,70 @@ describe(`helper-image`, () => {
-        assert.isTrue(options.ctx.restore.called);
-        assert.isTrue(options.ctx.rotate.calledWith(180 * Math.PI / 180));
-        assert.isTrue(options.ctx.scale.calledWith(-1, 1));
-        break;
-      case 5:
-        msg = `rotate270AndFlipImage`;
-        assert.isTrue(options.ctx.save.called);
-        assert.isTrue(options.ctx.translate.calledWith(options.x + options.width / 2, options.y + options.height / 2));
-        assert.isTrue(options.ctx.drawImage.calledWith(options.img, -options.width / 2, -options.height / 2, options.width, options.height));
-        assert.isTrue(options.ctx.restore.called);
-        assert.isTrue(options.ctx.rotate.calledWith(270 * Math.PI / 180));
-        assert.isTrue(options.ctx.scale.calledWith(-1, 1));
-        break;
-      case 6:
-        msg = `rotateImage270`;
-        assert.isTrue(options.ctx.save.called);
-        assert.isTrue(options.ctx.translate.calledWith(options.x + options.width / 2, options.y + options.height / 2));
-        assert.isTrue(options.ctx.drawImage.calledWith(options.img, -options.width / 2, -options.height / 2, options.width, options.height));
-        assert.isTrue(options.ctx.restore.called);
-        assert.isTrue(options.ctx.rotate.calledWith(2 * Math.PI - 270 * Math.PI / 180));
-        break;
-      case 7:
-        msg = `rotateNeg90AndFlipImage`;
-        assert.isTrue(options.ctx.save.called);
-        assert.isTrue(options.ctx.translate.calledWith(options.x + options.width / 2, options.y + options.height / 2));
-        assert.isTrue(options.ctx.drawImage.calledWith(options.img, -options.width / 2, -options.height / 2, options.width, options.height));
-        assert.isTrue(options.ctx.restore.called);
-        assert.isTrue(options.ctx.rotate.calledWith(90 * Math.PI / 180));
-        assert.isTrue(options.ctx.scale.calledWith(-1, 1));
-        break;
-      case 8:
-        msg = `rotateNeg90`;
-        assert.isTrue(options.ctx.save.called);
-        assert.isTrue(options.ctx.translate.calledWith(options.x + options.width / 2, options.y + options.height / 2));
-        assert.isTrue(options.ctx.drawImage.calledWith(options.img, -options.width / 2, -options.height / 2, options.width, options.height));
-        assert.isTrue(options.ctx.restore.called);
-        assert.isTrue(options.ctx.rotate.calledWith(2 * Math.PI - 90 * Math.PI / 180));
-        break;
-      default:
-        msg = `do nothing`;
-        break;
-      }
-      it(`${msg} on the canvas if image orientation is ${orientation}`, /* eslint no-empty-function: 0 */ () => {});
+    events.forEach(/* eslint complexity: ["error", 9] */ (def) => {
+      const {message, orientation} = def;
+      describe(`when an image file is received with image orientation as ${orientation}`, () => {
+        options.orientation = orientation;
+        file.image.orientation = orientation;
+        orient(options, file);
+        it(`${message} on the canvas`, /* eslint no-empty-function: 0 */ () => {
```

Comment:
I was really hoping we could avoid this switch statement. It still pretty unclear what's being proven.